### ISSUE TITLE
Possible to execute terminal command depending on user input?

### ISSUE THREAD

**Issue by Jones-S** [SBoudrias/Inquirer.js#856]
I would like to build a setup client, where the user can decide which plugins he wants to include into the project.

Is it somehow possible to execute a terminal command from inquirer?

Assuming a checkbox setup like:
(! The selection part is not working yet. I just took the checkbox example and added something to the `then` part.

```
/**
 * Checkbox list examples
 */

'use strict'
const { exec } = require('child_process')
const inquirer = require('inquirer')

inquirer
  .prompt([
    {
      type: 'checkbox',
      message: 'Select plugins',
      name: 'plugins',
      choices: [
        new inquirer.Separator(' = The Plugins = '),
        {
          name: 'Plugin 1 (susy)'
        },
        {
          name: 'Plugin 2'
        }
      ]
    }
  ])
  .then(answers => {
    console.log('Installing susy...')
    console.log('Exec', exec)

    exec('yarn add susy', (err, stdout, stderr) => {
      if (err) {
        console.log(err)
        // node couldn't execute the command
        return
      }

      // the *entire* stdout and stderr (buffered)
      console.log(`successfully installed susy`)
      console.log(`stdout: ${stdout}`)
      console.log(`stderr: ${stderr}`)
    })
  })
```

Is this even possible?
Because now the inquirer command just stops but does not end either...

Thanks for some pointers on this.
(BTW: I followed this: https://stackoverflow.com/a/20643568)

**Comment by Jones-S** [comment]
Oh wait. It did actually work. I just got no feedback in the terminal. After it was finished I got the output! Now I just need to figure out how to track the ongoing installation process.

**Comment by SBoudrias** [comment]
@Jones-S when you run exec or spawn, you need to proxy the output of the command to the node process stdout. https://nodejs.org/docs/latest/api/child_process.html

**Comment by Jones-S** [comment]
@SBoudrias Thank you for your input. In the code above I already have the `stdout` process. But only I can log the data after the command has finished. I would somehow need a way to log it continuously, not just all of it together when it's finished...

**Comment by colorfulWorld** [comment]
@Jones-S Did you solve it? I had the same problem

**Comment by Jones-S** [comment]
Well I have absolutely no idea anymore what I built back then. But I just digged into old code and found the things I was trying to build in 2019:
https://github.com/wearelucid/vuecid-nuxt-craft/blob/f9faafda12dd7cbbaea951b11d757ad8704e0b8a/gulpfile.js

Maybe this helps. I somehow managed to use user input to do different things...
But I don't know anymore if it was a workaround or if it actually solved my problem...

Hope you find some hints. Cheers
